src/vcBPA.cpp

```diff
@@ -899,23 +919,35 @@ vdkError vcBPA_ConvertOpen(vdkConvertCustomItem *pConvertInput, uint32_t everyNt
   vcBPA_Init(&pData->pManifold, pData->pContext);
   pData->pManifold->ballRadius = pData->ballRadius;
   pData->pManifold->gridSize = pData->gridSize;
+  pData->pManifold->pAttributes = &pData->attributes;
+  pData->pManifold->pMutex = udCreateMutex();
+  pData->pManifold->targetPoints = (uint32_t)pConvertInput->pointCount;
 
+  udDouble3 scale;
+  udDouble3 translation;
+  udDoubleQuat ypr;
   vdkPointCloudHeader header = {};
+
   vdkPointCloud_GetHeader(pData->pNewModel, &header);
   udDouble4x4 storedMatrix = udDouble4x4::create(header.storedMatrix);
-  udDouble3 startAABBCenter = (storedMatrix * udDouble4::create(header.pivot[0], header.pivot[1], header.pivot[2], 1.0)).toVector3();
+  storedMatrix.extractTransforms(translation, scale, ypr);
+  udDouble3 center = (storedMatrix * udDouble4::create(header.boundingBoxCenter[0], header.boundingBoxCenter[1], header.boundingBoxCenter[2], 1.0)).toVector3();
+  udDouble3 extents = scale * udDouble3::create(header.boundingBoxExtents[0], header.boundingBoxExtents[1], header.boundingBoxExtents[2]);
 
-  udDouble3 halfGrid = udDouble3::create(pData->pManifold->gridSize / 2.0);
-  pData->pManifold->grids.PushBack(vcBPAGrid::create(startAABBCenter, startAABBCenter - halfGrid, startAABBCenter + halfGrid));
+  for (uint8_t i = 1; i < 3; ++i)
```

Comment:
Just confirming, you're assuming `pData->shortAxis` is always `0`, so you're essentially skipping the first element?

Reply:
Yes